M2/Macaulay2/d/ctype.d

```diff
@@ -54,6 +54,7 @@ export isnewline  (c:char):bool := (chartype(c) & NEWLINE  ) != 0;
 export isquote    (c:char):bool := (chartype(c) & QUOTE    ) != 0;
 
 export isalnum  (s:string):bool := (
+     if int(uchar(s.0)) == 226 && length(s) == 3 then return true; -- unicode math symbol
```

Comment:
I'm not sure it's a good idea to change the definition of "isalnum", which has a specific meaning.  Also, we already allow unicode characters in symbols, as here:
```m2

i11 : 你好 = 1

o11 = 1

i12 : 你好

o12 = 1
```

So it doesn't seem necessary.

Reply:
unicode symbols are still allowed. the only extra rule is, unicodes starting with 0xe2 (which means mathematical symbols and a few others) can only be used stand-alone as symbols. your code above will run just fine.

Reply:
the alternative is to revert ctype.d to how it was before, and have an extra test when parsing in lex.d.
I thought it was uglier, but if you prefer that can be done.

Reply:
Well, "isalnum" should continue to check whether ascii strings are alphanumeric - that's what its name means.  Maybe you could do something different higher up.

I assume the goal to make `x≠y` work the way `x*y` works, which is a legitimate goal.

Reply:
that's right. right now `x≠y` is not considered alphanumeric. Is that the part you disagree with?

Reply:
No, I think we shouldn't call `≠` alphanumeric.  The adjective "alphanumeric" has a standard meaning and applies to strings of ascii characters -- it means each character is an alphabetic letter or a numeric digit.

Reply:
I see. before this PR, everything from 128 to 255 was considered ALPHA -- and a fortiori, ALNUM. what you're suggesting is a serious change in `ctype.d`. This PR had a much more modest goal.